src/snowflake/snowpark/functions.py

```diff
@@ -2742,7 +2742,7 @@ def char(col: ColumnOrName) -> Column:
     return builtin("char")(c)
 
 
-def to_char(c: ColumnOrName, format: Optional[ColumnOrLiteralStr] = None) -> Column:
```

Comment:
`ColumnOrLiteralStr` is referring to `lit("abc")` or `"abc"`, which is misleading since our type hints don't distinguish Column[str] or Column[int]. This is also inconsistent with functions like `to_binary`. Since local testing uses type hints to determine whether an expression should be evaluated as a column or a literal value, we should change this to `str.

Reply:
can users put a col expression as the `format`?
e.g., is the following allowed?
`to_char(col("data", format=col("format")))`

Reply:
I don't think so: https://docs.snowflake.com/en/sql-reference/functions/to_char